### ISSUE TITLE
Replay of Local Activities Executed by an Interceptor

### ISSUE THREAD

**Issue by grantfuhr** [temporalio/sdk-go#828]

**Comment by cretz** [comment]
Thanks for the report! At first glance I agree, this should not error. I will investigate.

**Comment by Lercher** [comment]
> The error you received commonly occurs if you have non-determinism.

Yes, I see: It's indeed the best explanation as we are working on implementations currently. I just was distracted by the nearly identical stack trace and no one mentioning non-determinism in this ticket. Sorry, my bad, and thank you very much.
